src/helpers/domainFindOptionsHelper.ts

```diff
@@ -62,77 +66,26 @@ export function createSort(req: HttpRequest): DynamicObject<SortDirection> {
         );
     }
   }
-  const sortObject = createSortDynamicObject(sortValues, sortDirection);
-
-  return sortObject;
-}
-
-/**
- * Creates a dynamic object with the variables named after the given string and the values set to the sortDirections.
- * Defaults to -1 when a direction is not supplied
- * @param sortValues - Array of strings to set as the variable names within the object
- * @param sortDirections - An array of sort directions to set the variables within the object to
- * @returns
- */
-export function createSortDynamicObject(
-  sortValues: string[],
-  sortDirections: SortDirection[]
-): DynamicObject<SortDirection> {
-  const sort: DynamicObject<SortDirection> = {};
-  sortValues = sortValues.filter((item, index) => {
-    return sortValues.indexOf(item) === index;
-  });
-  sortValues.forEach((x, index) => {
-    if (x === "created") {
-      const createdTimestamp = `${x}.timestamp`;
-      const createdBlockNumber = `${x}.blockNumber`;
-      const createdLogIndex = `${x}.logIndex`;
 
-      sort[createdTimestamp] =
-        sortDirections[index] !== undefined ? sortDirections[index] : -1;
-      sort[createdBlockNumber] =
-        sortDirections[index] !== undefined ? sortDirections[index] : -1;
-      sort[createdLogIndex] =
-        sortDirections[index] !== undefined ? sortDirections[index] : -1;
-    } else if (x === "buyNow.price") {
-      const listingPrice = "buyNow.value.listing.price";
-      // buyNow should sort by isActive desc
-      const isActive = "buyNow.value.isActive";
-
-      sort[listingPrice] =
-        sortDirections[index] !== undefined ? sortDirections[index] : -1;
-      sort[isActive] = -1;
-    } else if (x === "buyNow.time") {
-      const listingTimestamp = "buyNow.time.timestamp";
-      // buyNow should sort by isActive desc
-      const isActive = "buyNow.value.isActive";
-
-      sort[listingTimestamp] =
-        sortDirections[index] !== undefined ? sortDirections[index] : -1;
-      sort[isActive] = -1;
+  const sortObject: DynamicObject<SortDirection> = {};
+  sortValues.forEach((sortValue, index) => {
+    // Check if sort field is valid
+    if (sortValue in constants.mappingSortProps) {
+      // Map sort field to mutliple fields with default sort direction
+      for (const [mappingKey, defaultDirection] of Object.entries(
+        constants.mappingSortProps[sortValue]
+      )) {
+        sortObject[mappingKey] =
```

Comment:
How does this work with the nested properties in the mappingSortProps?

Reply:
Some sorting requires to pass multiple sorting fields with their directions.
i.e. `created` is required to sort by `created.timestamp`, `created.blockNumber`, `created.logIndex`.
`mappingSortProps` is to list all the valid sorting fields which can be used in API and resolve sorting fields to multiple fields in data-store.
And we can also find out invalid sorting fields from `mappingSortProps`.

After this `forEach` iteration, `created` will be resolved out like this:
```
sortObject["created.timestamp"] = -1;
sortObject["created.blockNumber"] = -1;
sortObject["created.logIndex"] = -1;
```

Reply:
So its sorting by all three of those fields at once? How can you  choose which one to make the primary sort

Reply:
How do we use primary and secondary sort keys?

Reply:
I think what maybe Ben is trying to get across is that you can't sort by `created` accurately unless you sort by these subprops in the **specific order** posted above.  So that's why `created` resolves to this result.